.github/workflows/main.yml

```diff
@@ -7,12 +7,30 @@ on:
 
 env:
   DOTNET_VERSION: 8.0.x
-  COVERALL_COLLECT_OUTPUT: "/p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/'"
-  COVERALL_MERGE_PATH: "/p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.json'"
 
 jobs:
 
+  Format:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: ${{ env.DOTNET_VERSION }}
+
+    - name: Check Format (*.cs)
+      run: dotnet format --verify-no-changes --verbosity diagnostic
+
+    - name: Build (Neo.CLI)
+      run: |
+        dotnet build ./src/Neo.CLI \
+        --output ./out/Neo.CLI
```

Comment:
Just a side comment: it might be a good idea to split this job into `Format` and `Build`, because `Format` job itself does not block the `Test` job. And then `Test` job may be dependent only on `Build` job. It's just the way how it works for NeoGo actions, and sometimes it's useful because your code may have a lack of formatting and you'll get your test results anyway.

Reply:
This is a required step/job. That is why its linked to tests. Or else @shargon would have to make the `build` and `format` job required for PRs.

Reply:
I think it's fine, or it can be done in a different PR, if the format is wrong we get the error earlier